src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -1457,12 +1513,38 @@ class MutateAllPartColumnsTask : public IExecutableTask
         bool is_full_part_storage = isFullPartStorage(ctx->new_data_part->getDataPartStorage());
         const auto & indices = ctx->metadata_snapshot->getSecondaryIndices();
 
+        bool secondary_indices_on_columns_alter_modify_drop = false;
+        bool secondary_indices_on_columns_alter_modify_rebuild = false;
+        NameSet alter_column_names;
+
+        MutationHelpers::getSecondaryIndicesOnColumnAlterModifyOptions(
+            ctx->commands,
+            ctx->data->getSettings(),
+            alter_column_names,
+            secondary_indices_on_columns_alter_modify_drop,
+            secondary_indices_on_columns_alter_modify_rebuild);
+
         MergeTreeIndices skip_indices;
         for (const auto & idx : indices)
         {
             if (removed_indices.contains(idx.name))
                 continue;
 
+            if (secondary_indices_on_columns_alter_modify_drop &&
+                std::any_of(idx.column_names.begin(),
+                            idx.column_names.end(),
+                            [&](String s) { return alter_column_names.contains(s); }))
+                continue;
+
+            if (secondary_indices_on_columns_alter_modify_rebuild &&
```

Comment:
It would be nice to integrate the code better with the existing code in l. 1548-1554.